rocketpy/rocket/components.py

```diff
@@ -63,27 +63,6 @@ def add(self, component, position):
         """
         self._components.append(self.component_tuple(component, position))
 
-    def get_by_type(self, component_type):
-        """Search the list of components and return a list with all the
-        components of the given type.
-
-        Parameters
-        ----------
-        component_type : type
-            The type of component to be returned.
-
-        Returns
-        --------
-        list
-            A list of components matching the specified type.
-        """
-        component_type_list = [
-            c.component
-            for c in self._components
-            if isinstance(c.component, component_type)
-        ]
-        return component_type_list
-
```

Comment:
Hey, I think this method could be useful to other applications rather than the monte carlo analysis. 

Btw what you did in here is a breaking change, formally speaking.

I recommend not removing it.